core/src/main/java/org/apache/accumulo/core/iterators/user/HardListIterator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.core.iterators.user;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.PartialKey;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.util.PeekingIterator;
+import org.apache.hadoop.io.Text;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * A wrapper making a list of hardcoded data into a SKVI. For testing.
+ */
+public class HardListIterator implements SortedKeyValueIterator<Key,Value> {
+  private static final Logger log = Logger.getLogger(HardListIterator.class);
+  public final static SortedMap<Key,Value> allEntriesToInject;
+  static {
+    SortedMap<Key,Value> t = new TreeMap<>();
+    t.put(new Key(new Text("a1"), new Text("colF3"), new Text("colQ3"), System.currentTimeMillis()), new Value("1".getBytes()));
+    t.put(new Key(new Text("c1"), new Text("colF3"), new Text("colQ3"), System.currentTimeMillis()), new Value("1".getBytes()));
+    t.put(new Key(new Text("m1"), new Text("colF3"), new Text("colQ3"), System.currentTimeMillis()), new Value("1".getBytes()));
+    allEntriesToInject = Collections.unmodifiableSortedMap(t); // for safety
+  }
+
+  private PeekingIterator<Map.Entry<Key,Value>> inner;
+  private Range seekRng;
+
+  @Override
+  public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
```

Comment:
Your comment made me realize HardListIterator was in core/src/main/java,
should probably be in core/src/test/java.
On Mar 27, 2015 7:33 PM, "Dylan Hutchison" <notifications@github.com> wrote:

> In
> core/src/main/java/org/apache/accumulo/core/iterators/user/HardListIterator.java
> <https://github.com/apache/accumulo/pull/22#discussion_r27338487>:
>
> > +public class HardListIterator implements SortedKeyValueIterator<Key,Value> {
> > +  private static final Logger log = Logger.getLogger(HardListIterator.class);
> > +  public final static SortedMap<Key,Value> allEntriesToInject;
> > +  static {
> > +    SortedMap<Key,Value> t = new TreeMap<>();
> > +    t.put(new Key(new Text("a1"), new Text("colF3"), new Text("colQ3"), System.currentTimeMillis()), new Value("1".getBytes()));
> > +    t.put(new Key(new Text("c1"), new Text("colF3"), new Text("colQ3"), System.currentTimeMillis()), new Value("1".getBytes()));
> > +    t.put(new Key(new Text("m1"), new Text("colF3"), new Text("colQ3"), System.currentTimeMillis()), new Value("1".getBytes()));
> > +    allEntriesToInject = Collections.unmodifiableSortedMap(t); // for safety
> > +  }
> > +
> > +  private PeekingIterator<Map.Entry<Key,Value>> inner;
> > +  private Range seekRng;
> > +
> > +  @Override
> > +  public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
>
> I confirmed that env.isFullMajorCompaction() is true when the tablet has
> no files on a standalone single-node instance.
>
> Writing a test for it is possible by writing an iterator, similar to
> HardListIterator, that emits an entry with a boolean value-- the result of
> env.isFullMajorCompaction(). It does not seem worth it to include a new
> iterator class in the source code for one test like this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/accumulo/pull/22/files#r27338487>.
>